<?xml version="1.0"?>
<launch>
  <arg name="display" default="true"/>

  <arg name="service"     default="true"/>
  <arg name="marker_type" default=""/>
  <arg name="sensor_type" default=""/>

  <arg name="sensor"      default=""/>
  <arg name="tf_prefix"   default="$(arg sensor)"/>

  <arg name="markers" default="markers"/>
  
  <!-- Mono and Depth Detector -->
  <arg name="image"       default=""/>
  <arg name="depth_image" default=""/>
  <arg name="camera_info" default=""/>
  <arg name="is_depth_in_meters" default="false" />

  <!-- Stereo Detector -->
  <arg name="image_left"    default=""/>
  <arg name="image_right"   default=""/>
  <arg name="stereo_info"   default=""/>

  <!-- Marker Parameters -->
  <arg name="dictionary"    default="3"/>

  <!-- corners to be consistent with checker board - squares minus one -->
  <arg name="board_width"   default="0"/><!-- corners to be consistent with checker board -->
  <arg name="board_height"  default="0"/><!-- corners to be consistent with checker board -->
  <arg name="square_length" default="0"/><!--mm-->
  <arg name="marker_length" default="0"/><!--mm-->

  <!-- Diamond parameters -->
  <arg name="centre"     default=""/>
  <arg name="top_left"   default=""/>
  <arg name="top_right"  default=""/>
  <arg name="bot_right"  default=""/>
  <arg name="bot_left"   default=""/>

  <group ns="$(arg sensor)">
      <node name="$(arg marker_type)_detector_node" pkg="aruco_detector" type="marker_detector_node" output="screen">
        <param name="display"       value="$(arg display)"/>

        <param name="service"       value="$(arg service)"/>
        <param name="marker_type"   value="$(arg marker_type)"/>
        <param name="sensor_type"   value="$(arg sensor_type)"/>

        <param name="tf_prefix"     value="$(arg tf_prefix)" />
        <param name="markers"       value="$(arg markers)"/>

        <param name="image"         value="$(arg image)" />
        <param name="depth_image"   value="$(arg depth_image)" />
        <param name="camera_info"   value="$(arg camera_info)" />
        <param name="is_depth_in_meters" type="bool" value="$(arg is_depth_in_meters)" />

        <param name="image_left"    value="$(arg image_left)" />
        <param name="image_right"   value="$(arg image_right)" />
        <param name="stereo_info"   value="$(arg stereo_info)" />

        <param name="dictionary"    value="$(arg dictionary)"/>
        <param name="board_width"    value="$(arg board_width)"/>
        <param name="board_height"   value="$(arg board_height)"/>
        <param name="marker_length"  value="$(arg marker_length)"/>
        <param name="square_length"  value="$(arg square_length)"/>
        
        <param name="centre"     value="$(arg centre)"/>
        <param name="top_left"   value="$(arg top_left)"/>
        <param name="top_right"  value="$(arg top_right)"/>
        <param name="bot_right"  value="$(arg bot_right)"/>
        <param name="bot_left"   value="$(arg bot_left)"/>
      </node>
   </group>
</launch>
